src/rustup-dist/src/manifestation.rs

```diff
@@ -161,13 +172,24 @@ impl Manifestation {
         }
 
         // Install components
-        for (component, installer_file) in things_to_install {
+        for (component, format, installer_file) in things_to_install {
 
             notify_handler(Notification::InstallingComponent(&component.pkg,
                                                              &self.target_triple,
                                                              component.target.as_ref()));
 
-            let package = try!(TarGzPackage::new_file(&installer_file, temp_cfg));
+            let gz;
+            let xz;
+            let package: &Package = match format {
+                Format::Gz => {
+                    gz = try!(TarGzPackage::new_file(&installer_file, temp_cfg));
+                    &gz
+                }
+                Format::Xz => {
+                    xz = try!(TarXzPackage::new_file(&installer_file, temp_cfg));
+                    &xz
+                }
+            };
```

Comment:
`rustc` does not complain, but I guess it might be regarded as bad practice.
It is easy to transform this in a single local that is always assigned by wrapping the two cases in an enum, if it is preferred.